src/backend/joanie/tests/core/test_models_enrollment.py

```diff
@@ -132,6 +134,29 @@ def test_models_enrollment_unique_opened_course_run_per_course_and_user(self):
             str(context.exception),
         )
 
+        enrollment.is_active = False
+        enrollment.save()
+
+        # If the first enrollment is not active anymore, user should be able to enroll
+        # to another course run for the same course
+        factories.EnrollmentFactory(
+            course_run=cr2, user=enrollment.user, is_active=True
+        )
+
+        # And finally it should not be able to re-enroll to the first course run
+        with self.assertRaises(ValidationError) as context:
+            factories.EnrollmentFactory(
+                course_run=cr1, user=enrollment.user, is_active=True
+            )
+
+        self.assertEqual(
+            (
+                "{'user': ['You are already enrolled to an opened course run "
+                f'for the course "{cr1.course.title}".\']}}'
+            ),
+            str(context.exception),
+        )
+
```

Comment:
It will be good to add a test with a date over(course_run__end) and to have two enrollments active but one with a date hover and to show it's possible in this case to register

Reply:
Good idea!